torch/csrc/jit/runtime/vararg_functions.cpp

```diff
@@ -39,6 +39,130 @@ void format(Stack& stack, size_t num_inputs) {
   push(stack, ss.str());
 }
 
+// IValue tags are intentionally private, so we need additional logic to cast
+// the IValue type to the specified format.
+void addFormattedArg(
+    char key,
+    const IValue& ival,
+    std::stringstream& ss,
+    int precision = 6) {
+  // TODO: Implement precison-based formatting
+  switch (key) {
+    case 'd':
+    case 'i': {
+      TORCH_CHECK(
+          ival.isScalar(),
+          "Got ",
+          ival.tagKind(),
+          ", but a number is required for formatting");
+      if (ival.isInt()) {
+        ss << ival.toInt();
+      } else {
+        ss << static_cast<int>(ival.toDouble());
+      }
+      break;
+    }
+    case 'e':
+    case 'E': {
+      TORCH_CHECK(
+          ival.isScalar(),
+          "Got ",
+          ival.tagKind(),
+          ", but a number is required for formatting");
+      ss << std::setprecision(precision) << std::scientific;
+      if (key == 'E') {
+        ss << std::uppercase;
+      }
+      if (ival.isInt()) {
+        ss << static_cast<float>(ival.toInt());
+      } else {
+        ss << static_cast<float>(ival.toDouble());
+      }
+      break;
+    }
+    case 'f':
+    case 'F': {
+      TORCH_CHECK(
+          ival.isScalar(),
+          "Got ",
+          ival.tagKind(),
+          ", but a number is required for formatting");
+      ss << std::setprecision(precision) << std::fixed;
```

Comment:
I don't think it matters *now*, but in general we should reset state like "precision" of printing stream when we are done in this case.